text_generation/causal_lm/cpp/continuous_batching/library/src/scheduler.hpp

```diff
@@ -277,8 +313,20 @@ class Scheduler {
                 size_t num_running_seqs = sequence_group->num_running_seqs();
                 // prompt phases can have a single running sequence
                 OPENVINO_ASSERT(num_running_seqs == 1);
-                // here we also assume that sequence must be scheduler in a single shot and has no already generated context
-                OPENVINO_ASSERT(sequence_group->get_context_len() == 0);
+                
+                size_t previously_allocated_blocks = 0;
+                // can happen due to partial preemprion
+                if (sequence_group->get_context_len()>0) {
```

Comment:
interesting case.. 
- do we have similar situation with `dynamic_split_fuse`? I hope no, because `dynamic_split_fuse` assumes we can partially compute prompt chunk by chunk 
- if it's vLLM specific, maybe we perform full recompute in vLLM case? to simplify current logic.

Reply:
Tested this logic for both `vLLM` and `dynamic_split_fuse`. 

It appeared that `vLLM` case had a bug in case of partial preemption of prompt. Allocated blocks that store partial prompt could be preempted, but not freed. Fixed it in 366-377 lines in scheduler.hpp.

`dynamic_split_fuse` works. Checked partial preemption of prompt, no errors occurred and generated text matches the case without preemption.

Reply:
as I remember, we agreed to fully preempt prompts for vLLM case to avoid changing this logic.

Reply:
Looks like I misunderstood. Changed to full preemption of prompts for vLLM case.